wgctrl-rs/src/device.rs

```diff
@@ -103,14 +109,137 @@ pub struct DeviceInfo {
     pub(crate) __cant_construct_me: (),
 }
 
+type RawInterfaceName = [c_char; libc::IFNAMSIZ];
+
+/// The name of a Wireguard interface device.
+#[derive(PartialEq, Eq, Clone, Copy)]
+pub struct InterfaceName(RawInterfaceName);
+
+impl TryFrom<&str> for InterfaceName {
+    type Error = InvalidInterfaceName;
+
+    fn try_from(name: &str) -> Result<Self, InvalidInterfaceName> {
+        let len = name.len();
+        // Ensure its short enough to include a trailing NUL
+        if len > (libc::IFNAMSIZ - 1) {
+            return Err(InvalidInterfaceName::TooLong(len));
+        }
+
+        if len == 0 || name.trim_start_matches('\0').is_empty() {
+            return Err(InvalidInterfaceName::Empty);
+        }
+
+        let mut buf = [c_char::default(); libc::IFNAMSIZ];
+        // Check for interior NULs and other invalid characters.
+        for (out, b) in buf.iter_mut().zip(name.as_bytes()[..(len - 1)].iter()) {
+            if *b == 0 {
+                return Err(InvalidInterfaceName::InteriorNul);
+            }
+
+            if *b == b'/' || *b == b' ' {
```

Comment:
```suggestion
            if *b == b'/' || b.is_ascii_whitespace() {
```

It looks like iproute2, for example, checks not just for a space character but any whitespace character via [`isspace(3)`](https://linux.die.net/man/3/isspace). We probably want to follow suit.

Reply:
Hadn't considered that but its probably smart to follow the C as closely as possible.